code.ts

```diff
@@ -1,21 +1,65 @@
+const clientStrageKey = 'team-library-components'
+
 async function main() {
-  if (figma.command == "saveTargetComponent") {
-    await saveTargetComponent()
+  if (figma.command == "saveComponent") {
+    await figma.clientStorage.setAsync(clientStrageKey, await saveComponent([figma.currentPage])) 
   }
   else if (figma.command == "replaceNodes") {
-    await replaceNodes()
+    const teamLibraryComponents = await figma.clientStorage.getAsync(clientStrageKey)    
+    await findTargetNodes(figma.currentPage.selection, teamLibraryComponents)
   }
   figma.closePlugin()
 }
 
-// ComponentFileのComponent IDを取得して保存
-async function saveTargetComponent() {
-  return 1
+async function saveComponent(nodes) {
+  const teamLibraryMasterComponents = {}
+  findComponent(teamLibraryMasterComponents, nodes)
+
+  return teamLibraryMasterComponents
+}
+
+async function findComponent(teamLibraryMasterComponents, nodes) {
+  for (const node of nodes) {
+    if (node.type === "COMPONENT") {
+      teamLibraryMasterComponents[node.name] = node.key
+    } 
+    if (node.children != null) {
+      findComponent(teamLibraryMasterComponents, node.children)
+    }
+  } 
 }
 
 // 保存したComponent IDを使ってリプレイス
-async function replaceNodes() {
-  return 1
+async function findTargetNodes(nodes, teamLibraryComponents) {
```

Comment:
```suggestion
async function scanNodes(nodes, teamLibraryComponents) {
```

- findだと「見つけたものを返す」という振る舞いが想起されるので、変えたい
- 意味としては `scan and replace` みたいなものが適当そう
  - `scanAndReplaceNodes` でもいいけど
  - `scanNodes` でも順当かなという気がする（imo）
- ちなみにユーザからの見え方は変わらず `replace` もしくは `transport` なんかでいいと思う（コマンド名）
  - 再帰的に走査して置換してまわるのは内部の話なので